mpi4jax/cython/cuda_runtime_api.pxd

```diff
@@ -1,13 +1,59 @@
 cdef extern from "cuda_runtime_api.h":
     cdef enum cudaError:
-        cudaSuccess
+        cudaSuccess = 0
+        cudaErrorInvalidValue = 1
+        cudaErrorMemoryAllocation = 2
+        cudaErrorInitializationError = 3
+        cudaErrorCudartUnloading = 4
+        cudaErrorProfilerDisabled= 5
+        cudaErrorProfilerNotInitialized = 6
+        cudaErrorProfilerAlreadyStarted = 7
+        cudaErrorProfilerAlreadyStopped = 8
+        cudaErrorInvalidConfiguration = 9
+        cudaErrorInvalidPitchValue = 12
+        cudaErrorInvalidSymbol = 13
+        cudaErrorInvalidHostPointer = 16
+        cudaErrorInvalidDevicePointer = 17
+        cudaErrorInvalidTexture = 18
+        cudaErrorInvalidTextureBinding = 19
+        cudaErrorInvalidChannelDescriptor = 20
+        cudaErrorInvalidMemcpyDirection = 21
+        cudaErrorAddressOfConstant = 22
+        cudaErrorTextureFetchFailed = 23
+        cudaErrorTextureNotBound = 24
+        cudaErrorSynchronizationError = 25
+        cudaErrorInvalidFilterSetting = 26
+        cudaErrorInvalidNormSetting = 27
+        cudaErrorMixedDeviceExecution = 28
+        cudaErrorNotYetImplemented = 31
+        cudaErrorMemoryValueTooLarge = 32
+        cudaErrorStubLibrary = 34
+        cudaErrorInsufficientDriver = 35
+        cudaErrorCallRequiresNewerDriver = 36
+        cudaErrorInvalidSurface = 37
+        cudaErrorDuplicateVariableName = 43
+        cudaErrorDuplicateTextureName = 44
+        cudaErrorDuplicateSurfaceName = 45
+        cudaErrorDevicesUnavailable = 46
+        cudaErrorIncompatibleDriverContext = 49
+        cudaErrorMissingConfiguration = 52
+        cudaErrorPriorLaunchFailure = 53
+        cudaErrorLaunchMaxDepthExceeded = 65
+        cudaErrorNoDevice = 100
+        cudaErrorInvalidDevice = 101
+        cudaErrorDeviceNotLicensed = 102
+        cudaErrorStartupFailure = 127
+        cudaErrorInvalidKernelImage = 200
+        cudaErrorDeviceUninitialized = 201
+        cudaErrorMapBufferObjectFailed = 205
```

Comment:
What do we need all these for? We're only using `cudaSuccess`

Reply:
Ah, i cherry picked an extra commit from my branch...

I autogenerated this file when I needed to debug the OOM error we had some time ago.
I was converting the int type to string (though indeed that part is missing here...).

I think it would be more helpfull to print the error name other than only the code... 
do you agree?

Reply:
Yes, definitely. I think there is a function in the CUDA runtime for that.

Reply:
Ok, with the latest commit now it prints something like
```python 
Traceback (most recent call last):
  File "mpi4jax/cython/mpi_xla_bridge_gpu.pyx", line 122, in mpi4jax.cython.mpi_xla_bridge_gpu.mpi_allreduce
    checked_cuda_memcpy(in_buf, data, count, cudaMemcpyDeviceToHost)
  File "mpi4jax/cython/mpi_xla_bridge_gpu.pyx", line 66, in mpi4jax.cython.mpi_xla_bridge_gpu.checked_cuda_memcpy
    raise RuntimeError(mystr)
RuntimeError: cudaMemcpy failed with the following error:
	Error #2 cudaErrorMemoryAllocation: out of memory
```